File: branches/Medusa/Source/Fantasy.Jobs/Scheduling/WeeklyTrigger.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.Serialization;
using Fantasy.XSerialization;
using System.Globalization;

namespace Fantasy.Jobs.Scheduling
{
    [DataContract]
    [XSerializable("weeklyTrigger", NamespaceUri = Consts.ScheduleNamespaceURI)]
    public class WeeklyTrigger : Trigger 
    {
        [DataMember]
        [XAttribute("daysOfWeek")]
        public DayOfWeek[] DaysOfWeek { get; set; }

        [DataMember]
        [XAttribute("weeksInterval")]
        public int WeeksInterval { get; set; }

        public WeeklyTrigger()
        {
            this.DaysOfWeek = new DayOfWeek[0]; 
            WeeksInterval = 1;
        }
       
        public override TriggerType Type
        {
            get { return TriggerType.Weekly; }
        }

        protected override string TriggerTimeDescription
        {
            get
            {
                 DateTimeFormatInfo info = DateTimeFormatInfo.GetInstance(null);
                 DateTime time = new DateTime(this.StartTime.Ticks);
                return string.Format("At {0:t} every {1} of every {2} {3}, starting {4:g}.",time,
                    string.Join(",", DaysOfWeek.Select(d=>info.GetDayName(d))), WeeksInterval, 
                    WeeksInterval > 1 ? "weeks" : "week", this.StartBoundary);   
            }
        }
    }
}

File: Fantasy.Jobs/src/fantasy/jobs/scheduling/WeeklyTrigger.java
package fantasy.jobs.scheduling;

import java.text.DateFormatSymbols;
import java.text.SimpleDateFormat;

import org.apache.commons.lang3.time.FastDateFormat;


import fantasy.*;
import fantasy.collections.*;
import fantasy.xserialization.*;

@XSerializable(name = "weeklyTrigger", namespaceUri = fantasy.jobs.Consts.ScheduleNamespaceURI)
public class WeeklyTrigger extends Trigger
{
	/**
	 * 
	 */
	private static final long serialVersionUID = 5003088522985743724L;

	@XAttribute(name = "daysOfWeek")
	private DayOfWeek[] privateDaysOfWeek;
	public final DayOfWeek[] getDaysOfWeek()
	{
		return privateDaysOfWeek;
	}
	public final void setDaysOfWeek(DayOfWeek[] value)
	{
		privateDaysOfWeek = value;
	}

	@XAttribute(name = "weeksInterval")
	private int privateWeeksInterval;
	public final int getWeeksInterval()
	{
		return privateWeeksInterval;
	}
	public final void setWeeksInterval(int value)
	{
		privateWeeksInterval = value;
	}

	public WeeklyTrigger()
	{
		this.setDaysOfWeek(new DayOfWeek[0]);
		setWeeksInterval(1);
	}

	@Override
	public TriggerType getType()
	{
		return TriggerType.Weekly;
	}
	
	private static String[] _dayNames = new DateFormatSymbols().getWeekdays();

	@Override
	protected String getTriggerTimeDescription()
	{

		java.util.Date time = new java.util.Date(this.getStartTime().getMilliseconds());
		FastDateFormat tf = FastDateFormat.getTimeInstance(FastDateFormat.SHORT);
		SimpleDateFormat sf = new SimpleDateFormat();
		return String.format("At %1$s every %2$s of every %3$s %4$s, starting %$5s.",tf.format(time), StringUtils2.join(",", new Enumerable<DayOfWeek>(this.getDaysOfWeek()).select(new Selector<DayOfWeek, String>(){

			@Override
			public String select(DayOfWeek item) {
				return _dayNames[item.getValue()];
			}})), getWeeksInterval(), getWeeksInterval() > 1 ? "weeks" : "week", sf.format(this.getStartBoundary()));
	}
}